modules/tinymce/src/core/main/ts/api/html/Styles.ts

```diff
@@ -266,7 +267,7 @@ const Styles = (settings: StylesSettings = {}, schema?: Schema): Styles => {
             }
 
             // Convert RGB colors to HEX
-            if (!rgbaRegExp.test(value)) {
+            if (!rgbaRegExp.test(value) && value.toLowerCase() !== 'transparent') {
```

Comment:
I think there's a more fundamental problem here. If you look at the `rgbaRegExp` and this test, it's converting every value that doesn't begin with `rgba`. That's the problem, specifically whitelisting `transparent` won't help.

Reply:
If you look back at TinyMCE 5, where we had the behaviour customers want, it was more or less as I described.
https://github.com/tinymce/tinymce/blob/05989b03bf56d98aaf3522630b481616799c4dc8/modules/tinymce/src/core/main/ts/api/html/Styles.ts#L60
https://github.com/tinymce/tinymce/blob/05989b03bf56d98aaf3522630b481616799c4dc8/modules/tinymce/src/core/main/ts/api/html/Styles.ts#L302

Reply:
Yes, the rgba regex need some fix up, I have updated it to match all 4 values.

The actual cause of the issue is this function
```
RgbaColour.fromString(value).each((rgba) => {}

const fromString = (rgbaString: string): Optional<Rgba> => {
  if (rgbaString === 'transparent') {
    return Optional.some(rgbaColour(0, 0, 0, 0));
  }
 }
 ```
 It recognise `transparent` as a valid string, and convert it into `rgba(0,0,0,0)`, that gets passed into this function
 ```
 const rgbaToHexString = (color: string): string =>
  RgbaColour.fromString(color)
    .map(HexColour.fromRgba)
    .map((h) => '#' + h.value)
    .getOr(color);
```
`fromRgba` function converts colour into `rgb` colour (in this case #000000), hence why the colour black.

Reply:
Good sleuthing! I think the regex issue is not just that it lacked detail, but swapping the condition logic means that _everything_ which isn't `rgba()` is passed for conversion, rather than just `rgb()` colors as intended.

Removing the `transparent` line looks like a much better fix than adding the `transparent` exclusion here. It's been in the code since 2018, so the exact reason will be lost in an old CI server before it was moved to open source. I can only hope tests fail if it's really needed.

But either way, I think swapping back to an `rgb` check here (perhaps using the regex from `RgbaColour.ts`) is a good idea, since that's clearly the intent of the code.

Reply:
According to documentation, all colours except `rgba` are converted to hex, so the if statement on the regex should be valid.
https://www.tiny.cloud/docs/tinymce/latest/migration-from-6x/#force-hex-color

I'll remove the `transparent` line and see if CI reports anything anywhere.

Reply:
I guess we need to update the docs there since transparent was not converted into rgba in v6.

Reply:
That's documentation written based on the code. The intent, at least from my perspective, was to return to our v5 behaviour.